staff/jordi-pujol/e-commerce-protos/scripts/cart.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Abstraction of Cart
+ * 
+ * @method (add)
+ *  @param {Object} object
+ *  @throws {TypeError}
+ *  @returns {Number} Array length
+ * 
+ */
+
+function Cart() {
+    this.cartItems = [];
+}
+    
+    Cart.prototype.add = function (item){
+
+        if (!(item instanceof Product)) throw TypeError ('Should be an object');
+
+        this.cartItems.push(item)
+        return this.cartItems.length
+    };
+
+    Cart.prototype.totalPrice = function(){
+        var sumP = 0;
+        for (let i = 0; i< this.cartItems.length; i++){
+           sumP += this.cartItems[i].price
+        };
+        return sumP
+    };
+
+    Cart.prototype.numberOfItems = function (){
```

Comment:
You have chosen the complicated path! jejeje what you have done is not the same as returning the length?